rainier-core/src/main/scala/com/stripe/rainier/sampler/LeapFrog.scala

```diff
@@ -1,23 +1,77 @@
 package com.stripe.rainier.sampler
 
-private[sampler] case class LeapFrog(
-    nVars: Int,
-    initialHalfThenFullStep: Array[Double] => Unit,
-    twoFullSteps: Array[Double] => Unit,
-    finalHalfStep: Array[Double] => Unit
-) {
+private[sampler] case class LeapFrog(density: DensityFunction) {
   /*
   Params layout:
   array(0..(n-1)) == ps
   array(n..(n*2-1)) == qs
   array(n*2) == potential
   array(n*2+1) == stepSize
```

Comment:
I don't think we need `stepSize` to be part of the input array anymore, right? It can just be passed as a normal param to LeapFrog.

Reply:
We want to call LeapFrog with different stepSizes though so seems like we could pass it as an argument to the functions that use it rather than looking it up in the array.